src/network/networkd-dhcp4.c

```diff
@@ -523,6 +524,19 @@ static void dhcp4_handler(sd_dhcp_client *client, int event, void *userdata) {
                 case SD_DHCP_CLIENT_EVENT_EXPIRED:
                 case SD_DHCP_CLIENT_EVENT_STOP:
                 case SD_DHCP_CLIENT_EVENT_IP_CHANGE:
+
```

Comment:
So, I have not tested yet, but it seems ipv4ll will be started even if DHCP works fine after an address is expired once. no?

Reply:
yes right . we should call this when case is ```SD_DHCP_CLIENT_EVENT_STOP:```